llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp

```diff
@@ -5520,7 +5520,8 @@ static SDValue combineSTORE(SDNode *N, TargetLowering::DAGCombinerInfo &DCI,
     // Unfortunately, we can't do this in the legalizer because there is no
     // way to setOperationAction for an non-simple type.
     StoreSDNode *ST = cast<StoreSDNode>(N);
-    if (!ST->getValue().getValueType().isSimple())
+    if (!ST->getValue().getValueType().isSimple() ||
+        ST->getValue().getScalarValueSizeInBits() >= 256)
       return lowerSTOREVector(SDValue(ST, 0), DCI.DAG, STI);
```

Comment:
The reason this was added was to allow for custom lowering of i256 loads which weren't a MVT at the time. Since you're adding them I'd instead update this bit to add custom lowering for the larger types as well:

https://github.com/llvm/llvm-project/blob/3d3307ecd8bdd6d9af0d82245c5fc50e4d624a7a/llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp#L768-L774